presto-orc/src/test/java/com/facebook/presto/orc/OrcTester.java

```diff
@@ -353,19 +353,60 @@ private void testStructRoundTrip(Type type, List<?> values)
         }
 
         if (missingStructFieldsTestsEnabled) {
-            Type readType = rowType(type, type, type, type, type, type);
-            Type writeType = rowType(type, type, type);
+            testValuesMissingFromEnd(type, values);
 
-            List writeValues = values.stream()
-                    .map(OrcTester::toHiveStruct)
-                    .collect(toList());
+            //These test values missing from the a struct field that is not the last field
+            testExtraValuesInReader(type, values);
+            testExtraValuesInWriter(type, values);
+        }
+    }
 
-            List readValues = values.stream()
-                    .map(OrcTester::toHiveStructWithNull)
-                    .collect(toList());
+    private void testExtraValuesInReader(Type type, List<?> values) throws Exception
+    {
+        Type readType = namedRowType(type, "field_a", "field_b", "field_c", "field_d");
+        Type writeType = namedRowType(type, "field_a", "field_b", "field_d"); // missing field_c
 
-            assertRoundTrip(writeType, readType, writeValues, readValues, true);
-        }
+        List writeValues = values.stream()
+                .map(OrcTester::toHiveStruct)
+                .collect(toList());
+
+        List readValues = values.stream()
+                .map(value -> asList(value, value, null, value))
```

Comment:
I'm concerned that these tests won't catch reading the wrong fields, so I propose that we do something like: 
````
for i to values.lenght
  asList(
        i % 3 = 0 ? value : null,
        i % 3 = 1 ? value : null,
        null,
        i % 3 = 2 ? value : null)
``

Reply:
Agreed. I also think that there really isn't a need to run these tests in particular with different data types. The behavior should not depend on whether we are dealing with a struct of doubles or strings. I put them here because this is where the other tests are.

Is there a better place to put these tests? If not I'll make this change. Also, in this case is there a danger that we end up always having nulls? Maybe we can default to different and unique values instead of defaulting to a null?

Reply:
Take a look at `TestOrcWriter` it writes data and then reads it back in.  It looks like it would be easy to make a new test like that for struct type migrations.

Reply:
The new tests look good, so let's remove these ones.